.github/workflows/deploy.yml

```diff
@@ -0,0 +1,77 @@
+name: Deploy To Server
```

Comment:
It might be better to have two workflows. One for testing and one for deployment. The CI (test) workflow would only run on feature branches while the deployment workflow would be run when the pull request is merged to `main`

Reply:
Yes indeed. After I realized that yesterday, I created [a new PR to divide this workflow to 2 workflows](https://github.com/tednguyendev/nimble_challenge/pull/40) and I also put the URL of that second PR on the description of this PR. Sorry for the confusion!